### ISSUE TITLE
element.uploadFile does not work with some image upload components

### ISSUE THREAD

**Issue by harshajayaweeraXHJ** [puppeteer/puppeteer#5461]
I'm trying to upload an image in https://my.gumtree.com/postad but it doesn't' work and doesn't give any errors either.

This is how image component looks like. 

![image text](https://i.ibb.co/kJ7JfyC/Post-an-ad-Gumtree-com.png)

This is the html source for this component:

```html
<div id="html5_1e23hbscjgte1ni1goba4orb2_container" class="moxie-shim moxie-shim-html5" style="position: absolute; top: 0px; left: 8px; width: 130px; height: 100px; overflow: hidden; z-index: 0;"><input id="html5_1e23hbscjgte1ni1goba4orb2" type="file" style="font-size: 999px; opacity: 0; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;" multiple="" accept="image/jpeg,.jpg,.jpeg,image/png,.png"> </div>
```

This is the code i'm using. I tried both 

```
const fileInput = await page.$('input[type=file]');
await fileInput.uploadFile('./img1.jpeg');

```
and 

```
const inputUploadHandle = await page.$('input[type=file]');
inputUploadHandle.uploadFile(...['img1.jpeg']);

```

But doesn't work.
I also tried using full paths but no luck.

Can someone please point out what is the issue here?

**Comment by NikolaiT** [comment]
I am having the same issue. Anyone?

**Comment by msbhatt2018** [comment]
try changing input to div

**Comment by harshajayaweeraXHJ** [comment]
> try changing input to div

Tried it but does not work. Can you please try it on Gumtree. will give you the existing source code if required

**Comment by paullewis** [comment]
This should be fixed as of 532ae573d29063ffd081ed9ee340d71b25320dec as we changed the underlying implementation to use `DOM.setFileInputFiles`, which behaves a lot more as expected. Going to close this issue, but please let us know if it's not fixed.
